packages/xod-client/src/editor/actions.js

```diff
@@ -71,15 +72,14 @@ export const deselectAll = () => (dispatch, getState) => {
     type: ActionType.EDITOR_DESELECT_ALL,
     payload: {},
   });
-  if (!Selectors.getModeChecks(state).isDefault) {
-    dispatch(setMode(EDITOR_MODE.DEFAULT));
-  }
+
+  dispatch(setMode(EDITOR_MODE.DEFAULT));
 };
 
 export const selectEntity = R.curry(
   (entityType, id, dispatch, getState) => {
     const state = getState();
-    if (!Selectors.getModeChecks(state).isEditing) return;
+    if (Selectors.getMode(state) !== EDITOR_MODE.SELECTING) return;
```

Comment:
This one and lines in the `Patch.jsx` (like `if (isMiddleButtonPressed(event) || this.props.mode === EDITOR_MODE.PANNING) return;`) looks inconsistent: I expect to see all the checks are collected on the one layer.
I mean that this line isn't required until we check modes in the component (and we don't have to call this action at all), or check modes in the actions (that a redux-thunks). But the last solution could be used, cause some handlers don't call actions, but changes component state.